pkg/assets/scheduling.go

```diff
@@ -0,0 +1,105 @@
+package assets
+
+import (
+	"context"
+	"fmt"
+
+	embedded "github.com/openshift/microshift/assets"
+
+	sv1 "k8s.io/api/scheduling/v1"
+	scv1 "k8s.io/client-go/kubernetes/typed/scheduling/v1"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+var (
```

Comment:
I would put everything (vars, initialization', etc. under ApplyPriorityClasses)

Otherwise, the memory for the scheme and schedulingCodecs will remain allocated for the whole life of MicroShift, while it's not necessary anymore.

If we keep the initialization/setup/etc, under ApplyPriorityClasses, that memory will be freed.

Reply:
Changed